rpcq/_base.py

```diff
@@ -113,9 +79,33 @@ def replace(self, **kwargs):
         :param kwargs: The replaced fields.
         :return: A copy of self.
         """
-        d = self.asdict()
-        d.update(kwargs)
-        return type(self)(**d)
+        return replace(self, **kwargs)
+
+    def _extend_by_deprecated_fields(self, d):
```

Comment:
What is this for?

Reply:
Serialization of messages without deprecated fields is more-or-less a dump of their internal dictionary, which is keyed on slot names. Messages with deprecated fields have some of their slot names faked through setters/getters, so the same serialization strategy won't write information into deprecated keys. So, instead, this method gets used to store any extra strategy that has to happen for serializing deprecated keys.

Reply:
Is there an example anywhere?

Reply:
This Lisp specification
```
      (defmessage |A| ()
        ((|SlotA| :type :integer :required nil :deprecated-by |SlotB|)
         (|SlotB| :type :integer :required nil :deprecates |SlotA|)))
```
generates this Python definition
```
@dataclass(eq=False, repr=False)
class A(Message):
    SlotB: int = None

    SlotA: InitVar[Optional[int]] = None

    @property
    def SlotA(self):
        warn('SlotA is deprecated, use SlotB instead')
        return self.SlotB

    @SlotA.setter
    def SlotA(self, value):
        warn('SlotA is deprecated, use SlotB instead')
        self.SlotB = value

    def _extend_by_deprecated_fields(self, d):
        super()._extend_by_deprecated_fields(d)

        d.SlotA = d.SlotB

    def __post_init__(self, SlotA):
        if not isinstance(SlotA, property):
            if "SlotB" not in self.__dict__ or self.SlotB is None:
                warn('SlotA is deprecated, use SlotB instead')
                self.__dict__["SlotB"] = SlotA
```